apps/web/pages/teams/index.tsx

```diff
@@ -1,41 +1,47 @@
+import type { GetServerSidePropsContext } from "next";
 import { serverSideTranslations } from "next-i18next/serverSideTranslations";
 
 import { TeamsListing } from "@calcom/features/ee/teams/components";
 import Shell from "@calcom/features/shell/Shell";
 import { WEBAPP_URL } from "@calcom/lib/constants";
 import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { trpc } from "@calcom/trpc";
 import { Button } from "@calcom/ui";
 import { Plus } from "@calcom/ui/components/icon";
 
 import PageWrapper from "@components/PageWrapper";
 
+import { ssrInit } from "@server/lib/ssr";
+
 function Teams() {
   const { t } = useLocale();
+  const [user] = trpc.viewer.me.useSuspenseQuery();
+
+  const NewTeamButton = () => (
+    <Button
+      variant="fab"
+      StartIcon={Plus}
+      type="button"
+      href={`${WEBAPP_URL}/settings/teams/new?returnTo=${WEBAPP_URL}/teams`}>
+      {t("new")}
+    </Button>
+  );
   return (
     <Shell
       heading={t("teams")}
       hideHeadingOnMobile
       subtitle={t("create_manage_teams_collaborative")}
-      CTA={
-        <Button
-          variant="fab"
-          StartIcon={Plus}
-          type="button"
-          href={`${WEBAPP_URL}/settings/teams/new?returnTo=${WEBAPP_URL}/teams`}>
-          {t("new")}
-        </Button>
-      }>
+      CTA={!user.organizationId || user.organization.isOrgAdmin ? <NewTeamButton /> : undefined}>
       <TeamsListing />
     </Shell>
   );
 }
 
-export const getStaticProps = async () => {
-  return {
-    props: {
-      ...(await serverSideTranslations("en", ["common"])),
-    },
-  };
+export const getServerSideProps = async (context: GetServerSidePropsContext) => {
+  const ssr = await ssrInit(context);
+  await ssr.viewer.me.prefetch();
+
+  return { props: { trpcState: ssr.dehydrate(), ...(await serverSideTranslations("en", ["common"])) } };
```

Comment:
Replaced `useSession` with the new data fetching method.

Reply:
<img width="1011" alt="Screenshot 2023-07-07 at 11 52 51 AM" src="https://github.com/calcom/cal.com/assets/1780212/dea5d4dc-8a69-4ea7-a274-22130db4a281">
Don't think we need to call `serverSideTranslations` separately. ssrInit does that already.
```suggestion
export const getServerSideProps = async (context: GetServerSidePropsContext) => {
  const ssr = await ssrInit(context);
  await ssr.viewer.me.prefetch();

  return { props: { trpcState: ssr.dehydrate() } };
```